runtime/src/chpl-file-utils.c

```diff
@@ -113,6 +117,78 @@ qioerr chpl_fs_exists(int* ret, const char* name) {
   return err;
 }
 
+int chpl_fs_get_cuid(void) {
+  return getuid();
+}
+
+/*
+  Quick implementation of strcpy() using qio_malloc(). If it already exists
+  I can easily replace this.
+*/
+char* copy_string(const char* input);
```

Comment:
@mppf and/or @gbtitus - do you know if this exists already?  I'm not aware of a version that's already present but I also am not as familiar with the runtime or qio.

Reply:
Probably more feedback than expected, but ...

I'm not fond of this mainly because it's a bit of a mess in terms of interface.  I'd expect something called `chpl-file-utils` to provide file utilities, which string copying decidedly isn't part of.  A better place for this function would be chpl-string-support.  There is actually a `chpl_strcpy()` there, although it returns a `const char*` instead of plain `char*`.

A second bit of feedback: anything defined by Chapel in the public namespace should start with `chpl_`.  Just `copy_string` is likely to run into a user symbol someday.

Finally, why allocate via `qio_malloc()`?  Unless there's a good reason not to we should be getting memory from the Chapel allocator (chpl-mem.h).  And if there is a good reason to get system memory instead (which is what `qio_malloc()` happens to return currently), we should just call `sys_malloc()` directly, since there isn't any inherent connection between where qio might want to get memory from and where this string copier function might want to get it from.

Reply:
Are you looking for `qio_strdup`?

> Finally, why allocate via qio_malloc()? Unless there's a good reason not to we should be getting memory from the Chapel allocator (chpl-mem.h). And if there is a good reason to get system memory instead (which is what qio_malloc() happens to return currently), we should just call sys_malloc() directly, since there isn't any inherent connection between where qio might want to get memory from and where this string copier function might want to get it from.

`qio_malloc` does go through the chapel allocator, not the system one. When used in our runtime it's just an alias for `chpl_mem_alloc` (there is a branch for when the runtime is tested in a standalone fashion that maps it to sys_malloc, but that's not the normal case.)

Reply:
Well, that's amusing, that's the 2nd time today that standalone runtime stuff has led me to a wrong conclusion.  Sigh.

I still don't see a natural reason why either qio or chpl-file-utils should be supplying a string copy function, nor do I think `copy_string()` should be going around in public without a prefix on its name.

Reply:
Right, I'd expect you to use `chpl_strcpy`and you can cast the result if necessary.

Reply:
> Well, that's amusing, that's the 2nd time today that standalone runtime stuff has led me to a wrong conclusion. Sigh.

Is there some way we could make it more obvious? Or handle the C-only testing differently? (Maybe link the whole Chapel runtime, somehow?)

Reply:
> Is there some way we could make \[the standalone runtime stuff\] more obvious?

Perhaps.   I'm musing on that now.  For the time being I'm just going to try to be more context-aware.